src/ParallelAlgorithms/Events/ObserveAtExtremum.hpp

```diff
@@ -244,6 +244,21 @@ ObserveAtExtremum<ObservationValueTag, tmpl::list<ObservableTensorTags...>,
                      << scalar_name << "' is not known. Known tensors are: "
                      << ((db::tag_name<ObservableTensorTags>() + ",") + ...));
   }
+
+  tmpl::for_each<tmpl::list<ObservableTensorTags...>>(
+      [this, &context](auto tag_v) {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        const std::string tensor_name = db::tag_name<tag>();
+        if (tensor_name == scalar_name) {
+          if constexpr (tt::is_a_v<std::optional, typename tag::type>) {
+            PARSE_ERROR(context,
+                        "Cannot observe extremum of optional at this point");
```

Comment:
I think a simple check of `tag::type::value_type::rank()` would do the job here.

Reply:
That seems to work, thanks!